java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java

```diff
@@ -260,49 +418,303 @@ public void save(Dataset<Row> featureData, Map<String, String> writeOptions)
     }
   }
 
+  /**
+   * Persist the metadata and materialize the feature group to the feature store or insert data from a dataframe into
```

Comment:
This first sentence is a bit hard to read. It's also explained a bit later in the documentation that if the feature group doesn't exists the metadata is created, so I'm not sure we need this first sentence. 

Same applies to all the other `insert()` variations.